lib/msf/core/post/windows/services.rb

```diff
@@ -254,15 +309,17 @@ def service_info(name)
     end
 
     servicekey = "HKLM\\SYSTEM\\CurrentControlSet\\Services\\#{name.chomp}"
-    service[:display]     = registry_getvaldata(servicekey,"DisplayName").to_s
-    service[:starttype]   = registry_getvaldata(servicekey,"Start").to_i
-    service[:path]        = registry_getvaldata(servicekey,"ImagePath").to_s
-    service[:startname]   = registry_getvaldata(servicekey,"ObjectName").to_s
-    service[:dacl]        = nil
-    service[:logroup]     = nil
-    service[:interactive] = nil
-
-    return service
+    start_type = registry_getvaldata(servicekey, 'Start').to_s
+
+    {
+      display: registry_getvaldata(servicekey, 'DisplayName').to_s,
+      starttype: (start_type.starts_with?('0x') ? start_type.to_i(16) : start_type.to_i),
```

Comment:
There is a potential issue here if the string returned from `registry_getvaldata` is empty as you will end up setting `starttype` to `0` as a result, which will not match any service types.

```
irb(main):001:0> nil.to_s
=> ""
irb(main):002:0> "".to_i
=> 0
irb(main):003:0> 
```

Reply:
Resolved with https://github.com/rapid7/metasploit-framework/pull/16928/commits/51b6d0b596eb5ff1f9ff7f64e196146ac743eb98